### PR TITLE
Convert the InfoPanel and Topology components to new watcher data structure

### PR DESCRIPTION
**Pull request by hatched**
## Done

- Converted the InfoPanel and Topology components to use the new watcher data structure.
- Added a delta handler for the annotation delta events.

## QA

Make sure that the side InfoPanel is shown with the topology.

### CODE REVIEW

File: src/components/InfoPanel/InfoPanel.test.js [hunk 1]
```diff
@@ -1,29 +1,37 @@
+import { render, screen } from "@testing-library/react";
 import configureStore from "redux-mock-store";
 import { Provider } from "react-redux";
-import { mount } from "enzyme";
```
Comment (anthonydillon) [inline src/components/InfoPanel/InfoPanel.test.js:3]: Was it difficult to make the switch?

File: src/components/InfoPanel/InfoPanel.test.js [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
+import { reduxStateFactory } from "testing/redux-factory";
 import TestRoute from "components/Routes/TestRoute";
 
 import InfoPanel from "./InfoPanel";
 
 const mockStore = configureStore([]);
 
 jest.mock("components/Topology/Topology", () => {
-  const Topology = () => <div className="topology"></div>;
+  const Topology = () => <div data-testid="topology"></div>;
```
Comment (anthonydillon) [inline src/components/InfoPanel/InfoPanel.test.js:15]: I believe the standard is `data-test`

File: src/components/InfoPanel/InfoPanel.test.js [hunk 3]
```diff
@@ -32,16 +40,24 @@ describe("Info Panel", () => {
...
+            { name: "enterprise", owner: "kirk@external", type: "kubernetes" },
+          ],
+        },
+      }
+    );
+    const store = mockStore(mockState);
+    render(
       <Provider store={store}>
-        <MemoryRouter
-          initialEntries={["/models/user-eggman@external/group-test"]}
```
Comment (anthonydillon) [inline src/components/InfoPanel/InfoPanel.test.js:43]: Bye bye user-eggman :wave:

**Review by anthonydillon — COMMENTED** [review]
Few comments in the code

File: src/components/InfoPanel/InfoPanel.test.js (hunk 1 above)
Comment (hatched) [reply src/components/InfoPanel/InfoPanel.test.js:3]: Honestly it was trivial. The only issue being that we have to have a mindset change from querying by class name to things like `role`, `text`, or the escape hatch of `testid`.

File: src/components/InfoPanel/InfoPanel.test.js (hunk 2 above)
Comment (hatched) [reply src/components/InfoPanel/InfoPanel.test.js:15]: Yeah that is(was?) the standard but RTL has a method `getByTestId` in which it requires `data-testid` vs `data-test`.

File: src/components/InfoPanel/InfoPanel.test.js (hunk 3 above)
Comment (hatched) [reply src/components/InfoPanel/InfoPanel.test.js:43]: Poor eggman, he has been dethroned by Captain Kirk.

**Comment by anthonydillon** [comment]
You are missing provider from the infopanel:
image

File: src/components/InfoPanel/InfoPanel.test.js (hunk 2 above)
Comment (anthonydillon) [reply src/components/InfoPanel/InfoPanel.test.js:15]: I see, good to know.

Comment (anthonydillon) [reply src/components/InfoPanel/InfoPanel.test.js:15]: Better to follow the testing frameworks standard then ours

**Review by anthonydillon — CHANGES_REQUESTED** [review]
QA issue above

**Comment by hatched** [comment]
Oh good catch :D I actually removed that as it's an implementation detail of Juju. It's dependent on the cloud that you're on so the user knowing what the provider is called is irrelevant. 

The data is also not available in the watcher so we'd have to make extra requests for it.

**Comment by anthonydillon** [comment]
+1 after a discussion on mm

**Review by anthonydillon — APPROVED** [review]

**Comment by hatched** [comment]
Thanks, much appreciated 👍

**Pull request merged**